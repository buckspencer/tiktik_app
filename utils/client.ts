import sanityClient from '@sanity/client';

export const client = sanityClient({
  projectId: 'xz046p2y',
  dataset: 'production',
  apiVersion: '2022-06-05',
  useCdn: false,
  token: process.env.NEXT_PUBLIC_SANITY_TOKEN,
});
