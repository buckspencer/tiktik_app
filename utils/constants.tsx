import { BsCode, BsEmojiSunglasses } from "react-icons/bs";
import { FaGamepad, FaMedal, FaPaw } from "react-icons/fa";
import { GiCakeSlice, GiGalaxy, GiLipstick } from "react-icons/gi";

import { Reaction } from "@charkour/react-reactions";

export const reactionEmojis = {
  reactionThumbsUp: "👍",
  reactionThumbsDown: "👎",
  reactionParty: "🎉",
  reactionSmile: "😄",
  reactionHeart: "❤️",
  reactionFrown: "😕",
};

export const selectorEmojis: Reaction[] = [
  { label: "yup", node: <div>👍</div>, key: "reactionThumbsUp" },
  { label: "nope", node: <div>👎</div>, key: "reactionThumbsDown" },
  { label: "party", node: <div>🎉</div>, key: "reactionParty" },
  { label: "smile", node: <div>😄</div>, key: "reactionSmile" },
  { label: "heart", node: <div>❤️</div>, key: "reactionHeart" },
  { label: "hmm", node: <div>😕</div>, key: "reactionFrown" },
];

export const topics = [
  {
    name: "development",
    icon: <BsCode />,
  },
  {
    name: "comedy",
    icon: <BsEmojiSunglasses />,
  },
  {
    name: "gaming",
    icon: <FaGamepad />,
  },
  {
    name: "food",
    icon: <GiCakeSlice />,
  },
  {
    name: "dance",
    icon: <GiGalaxy />,
  },
  {
    name: "beauty",
    icon: <GiLipstick />,
  },
  {
    name: "animals",
    icon: <FaPaw />,
  },
  {
    name: "sports",
    icon: <FaMedal />,
  },
];

export const footerList1 = [
  "About",
  "Newsroom",
  "Store",
  "Contact",
  "Careers",
  "ByteDance",
  "Creator Directory",
];
export const footerList2 = [
  "qowturn for Good",
  "Advertise",
  "Developers",
  "Transparency",
  "qowturn Rewards",
];
export const footerList3 = [
  "Help",
  "Safety",
  "Terms",
  "Privacy",
  "Creator Portal",
  "Community Guidelines",
];
