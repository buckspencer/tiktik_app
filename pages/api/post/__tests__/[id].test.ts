import type { PageConfig } from 'next';
import { client } from '../../../../utils/client';
import endpoint from '../[id]';
import { testApiHandler } from 'next-test-api-route-handler';

const handler: typeof endpoint & { nextConfig?: PageConfig } = endpoint;

describe("GET", () => {
  it("should return a post", async () => {
    await testApiHandler({
      handler,
      params: { id: 'cW1QlAWGawhhxvB3TIW68X' },
      test: async ({ fetch }) => {
        const res = await fetch({ method: "GET" });

        await expect(res.json()).resolves.toMatchSnapshot();
      }
    });
  });
});
