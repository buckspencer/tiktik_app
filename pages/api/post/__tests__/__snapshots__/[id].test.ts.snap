// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET should return a post 1`] = `
Object {
  "_id": "cW1QlAWGawhhxvB3TIW68X",
  "caption": "Upload with uppy",
  "comments": Array [
    Object {
      "_key": "e8608d20-1ba3-4cb1-92fe-8e3c0f59f795",
      "comment": "We need more websites.",
      "postedBy": Object {
        "_id": "111003577416079876666",
        "_ref": null,
      },
    },
    Object {
      "_key": "d4aee97d-a9f2-4d5c-bf2d-de15525eb207",
      "comment": "Wow ",
      "postedBy": Object {
        "_id": "111003577416079876666",
        "_ref": null,
      },
    },
    Object {
      "_key": "e713e315-2fc5-457c-9e93-2131d20c96d2",
      "comment": "Add for test. ",
      "postedBy": Object {
        "_id": "111003577416079876666",
        "_ref": null,
      },
    },
    Object {
      "_key": "b7ac06c3-d169-4168-8551-a75289eec157",
      "comment": "wonderful",
      "postedBy": Object {
        "_id": "111003577416079876666",
        "_ref": null,
      },
    },
    Object {
      "_key": "863b3755-6f88-44c8-834d-4e3665cf9531",
      "comment": " FOO BAR",
      "postedBy": Object {
        "_id": "111003577416079876666",
        "_ref": null,
      },
    },
  ],
  "likes": Array [],
  "postedBy": Object {
    "_id": "111003577416079876666",
    "image": "https://lh3.googleusercontent.com/a-/AFdZucq8A73XNM5uirfCwK6VVgdUboDra2ryeQWwRA12q-w=s96-c",
    "userName": "Buck Spencer",
  },
  "reactions": Object {
    "reactionFrown": Array [
      "111003577416079876666",
    ],
    "reactionHeart": Array [],
    "reactionParty": Array [
      "111003577416079876666",
    ],
    "reactionSmile": Array [],
    "reactionThumbsDown": Array [],
    "reactionThumbsUp": Array [
      "111003577416079876666",
    ],
  },
  "userId": "111003577416079876666",
  "video": Object {
    "asset": Object {
      "_id": "file-72308b8d8233f363e0e11a82b0025e63a59c01ae-mp4",
      "url": "https://cdn.sanity.io/files/xz046p2y/production/72308b8d8233f363e0e11a82b0025e63a59c01ae.mp4",
    },
  },
}
`;

exports[`PUT insert should return post with newly inserted comment 1`] = `
Object {
  "id": "cW1QlAWGawhhxvB3TIW68X",
}
`;
